docs/griptape-framework/structures/task-memory.md

```diff
@@ -1,89 +1,320 @@
+                             Actions: [{"tag": "retrieve_sqrt", "name": "TaskMemoryClient", "path": "query", "input": {"values": {"memory_name": "TaskMemory",
+                             "artifact_namespace": "7554b69e1d414a469b8882e2266dcea1", "query": "What is the result of the calculation?"}}}]
+[04/26/24 13:13:16] INFO     Subtask 32b9163a15644212be60b8fba07bd23b
+                             Response: The result of the calculation is 111.1080555135405.
+[04/26/24 13:13:17] INFO     ToolkitTask 5b46f9ef677c4b31906b48aba3f45e2c
+                             Output: The square root of 12345 is approximately 111.108.
+```
+
+Note that on the `TaskMemoryClient` we've set `off_prompt` to `False` so that the results of the query can be returned directly to the LLM. If we had kept it as `True`, the results would have been stored back Task Memory which would've put us back to square one.
 
+While this fixed the problem, it took a handful more steps than when we just had `Calculator(off_prompt=False)`. Let's look at a more complex example where Task Memory shines.
 
-## Task Memory
-Here is an example of how memory can be used in unison with multiple tools to store and load content:
+## Large Data
+
+Let's say we want to query the contents of a very large webpage.
 
 ```python
-from griptape.artifacts import TextArtifact, BlobArtifact
-from griptape.memory.task.storage import TextArtifactStorage, BlobArtifactStorage
 from griptape.structures import Agent
-from griptape.tools import WebScraper, FileManager, TaskMemoryClient
-from griptape.engines import VectorQueryEngine, PromptSummaryEngine, CsvExtractionEngine, JsonExtractionEngine
-from griptape.drivers import LocalVectorStoreDriver, OpenAiEmbeddingDriver, OpenAiChatPromptDriver
+from griptape.tools import WebScraper
+
+# Create an agent with the Calculator tool
+agent = Agent(tools=[WebScraper(off_prompt=False)])
+
+agent.run(
+    "According to this page https://en.wikipedia.org/wiki/Elden_Ring, how many copies of Elden Ring have been sold?"
+)
+```
+
+When running this example, we get the following error:
+```
+[04/26/24 13:20:02] ERROR    ToolkitTask 67e2f907f95d4850ae79f9da67df54c1
+                             Error code: 400 - {'error': {'message': "This model's maximum context length is 8192 tokens. However, your messages resulted in 73874 tokens.
+                             Please reduce the length of the messages.", 'type': 'invalid_request_error', 'param': 'messages', 'code': 'context_length_exceeded'}}
+```
+
+This is because the content of the webpage is too large to fit in the LLM's input token limit. We can fix this by storing the content in Task Memory, and then querying it with the `TaskMemoryClient`.
+
+```python
+from griptape.structures import Agent
+from griptape.tools import WebScraper, TaskMemoryClient
+
+agent = Agent(
+    tools=[
+        WebScraper(off_prompt=True),
+        TaskMemoryClient(off_prompt=False),
```

Comment:
We may have covered this already in a separate conversation, but what is the value of `TaskMemoryClient` having `off_prompt` at all? would it every be true?

Reply:
Yes, the Sensitive Data example has it set to `true`. Will add more details.